mobilecoind-json/src/bin/main.rs

```diff
@@ -267,6 +288,79 @@ fn transfer(
     Ok(Json(JsonTransferResponse::from(&resp)))
 }
 
+/// Performs a transfer from a monitor and subaddress. The public keys and amount are in the POST data.
+#[post(
+    "/monitors/<monitor_hex>/subaddresses/<subaddress_index>/generate-tx",
+    format = "json",
+    data = "<generate>"
+)]
+fn generate_tx(
+    state: rocket::State<State>,
+    monitor_hex: String,
+    subaddress_index: u64,
+    generate: Json<JsonGenerateTxRequest>,
+) -> Result<Json<JsonGenerateTxResponse>, String> {
+    let monitor_id =
+        hex::decode(monitor_hex).map_err(|err| format!("Failed to decode monitor hex: {}", err))?;
+
+    let public_address = PublicAddress::try_from(&generate.transfer.receiver)?;
+
+    // Generate an outlay
+    let mut outlay = mc_mobilecoind_api::Outlay::new();
```

Comment:
Why doesn't this match the RPC GenerateTx that supports multiple outlays?

Reply:
I don't want to preconstruct the outlays and pass them to the request, so I think maybe I should rename the call to "generate_tx_proposal?"

Reply:
Renamed to `generate_request_code_transaction` to reflect difference from mobilecoind gRPC API